crates/dht/src/dht/snapshot.rs

```diff
@@ -0,0 +1,763 @@
+//! A snapshot of the DHT state at a given point in time.
+//!
+//! This module is public because its types need to be communicated between DHT instances, but it is
+//! largely opaque to the user. See [crate::dht::Dht::snapshot_minimal] and
+//! [crate::dht::Dht::handle_snapshot] for more information about using this module.
+
+use kitsune2_api::Timestamp;
+use std::collections::{HashMap, HashSet};
+
+/// A snapshot of the DHT state at a given point in time.
+///
+/// This is largely opaque to the user of the [crate::dht::Dht] model. It is intended to be sent
+/// between nodes to compare their DHT states and compared with [DhtSnapshot::compare].
+#[derive(Debug, Eq, PartialEq)]
+pub enum DhtSnapshot {
```

Comment:
`Disc -> Past, Ring -> Recent`

Consider this a very light suggestion. I like that we already have terminology similar to this in legacy Kitsune. But, I'm actually fine with the Disc/Ring names as long as we document them well and are consistent about them.

Other things I considered:
- `Disc -> Old, Ring -> New` - Working with data, I like to use the prefix `old_`, and wouldn't want to see `old_old_`.
- `Disc -> Historical? History?, Ring -> Recent` - Past is shorter.

```rust
pub enum DhtSnapshot {
    Minimal {
        past_top_hash: bytes::Bytes,
        past_boundary: Timestamp,
        recent_top_hashes: Vec<bytes::Bytes>,
    },
    PastSectors {
        past_sector_top_hashes: HashMap<u32, bytes::Bytes>,
        past_boundary: Timestamp,
    },
    PastSectorDetails {
        past_sector_hashes: HashMap<u32, HashMap<u64, bytes::Bytes>>,
        past_boundary: Timestamp,
    },
    RecentSectorDetails {
        recent_sector_hashes: HashMap<u32, HashMap<u32, bytes::Bytes>>,
        past_boundary: Timestamp,
    },
}
```

Reply:
```
         |-----Past-----|Recent-|
u32::MAX +----+----+----+--+--+-+-------
         |    |    |    |  |  | | Sector
L        +----+----+----+--+--+-+-------
o        |    |    |    |  |  | |
c        +----+----+----+--+--+-+
a        |    |    |    |  |  | |
t        +----+----+----+--+--+-+
i        |    |    |    |  |  | |
o      0 +----+----+----+--+--+-+
n        0         Time         Now
```

Reply:
I'd be happy with Historical/Recent. As you say, that's similar to terminology we already have and they make a nice pair.

The bit I'd be losing slightly and have to come up with some other wording for is that the Historical starts out singular because we compress that right down to one hash. The Recent starts out pluralized because it's a more predictable size and we want to diff it in fewer steps. I've already had to do that with adding `_sectors` and then `_sector_details`, so that's nothing new. Just needs a bit of thought.

Another option, because ring is not a geometry term and that's part of the confusion here because it's too similar to disc. The correct term would be [annulus](https://en.wikipedia.org/wiki/Annulus_(mathematics))... But I think there's some potential for misspellings and slips of the tongue trying to get the plural "annuli" right :)

Reply:
I'll wait for some more votes for options here before I start a big rename, it'll be a bit of a project :)

Reply:
> ring is not a geometry term

It is in German...it's called "[Kreisring](https://de.wikipedia.org/wiki/Kreisring)". Maybe just just need to add somewhere in the inline docs that it's the thing that's supposed to be thrown into volcanoes rather than the thing that keeps Elvis alive if you put it on a record player ;).

Reply:
That's a much better name, a "circle ring"!

Technically a top hash over the set of rings would represent a one ring... :D

Reply:
I think I'm derailing too far into non-technical terminology but it could also be considered as "flat donut" vs. a "frisbee". Or a 2d-torus?

Reply:
I'm fine with disc/ring, or past/recent. Another option would that is maybe more consistent would be full/partial. The comments make it clear enough that I don't feel strongly. I assuming these terms are contained within this module anyway, right? Or will they leak out?

Reply:
They're visible as enum variants but nothing needs to operate on them, just check for a "*Details" enum variant and cache that but otherwise the enum doesn't need to be inspected. The fields will have to be sent over gossip so the gossip protocol will use the sane names... it leaks :)